### PR TITLE
Use env var ENGI__DISABLEGITHUBINTEGRATION to skip GitHub checks

### PR DESCRIPTION
**Pull request by cck197**
I want to skip the GitHub checks for end-to-end integration testing. 

With this change, via a Python CLI `pytest` test I can:
1. Submit an analysis to the `analysis.submit` mutation
2. See the analysis request come back via the SNS -> SQS fanout
3. Process the request in `app.py`
4. Verify the appropriate changes in RavenDB

I couldn't figure out the ASP.NET `appsettings.json` interface. You'll probably want to eliminate my environment variable and use the configuration provider instead. Although, I'm not a big fan of the language-specific hierarchical configuration files as it's nice to [manage all the config in one place](https://12factor.net/config).

### CODE REVIEW

File: engi-server/Types/Analysis/AnalysisMutations.cs [hunk 1]
```diff
@@ -45,42 +45,49 @@ public AnalysisMutations()
...
         var (enrollment, repoReference) = user.GithubEnrollments.Find(repositoryFullName);
 
-        if (enrollment == null)
-        {
-            throw new ExecutionError("User does not have access to repository.") { Code = "FORBIDDEN" };
-        }
-
-        Repository repository;
-        GitHubCommit commit;
+        var skipGitHub = Environment.GetEnvironmentVariable("ENGI__DISABLEGITHUBINTEGRATION");
```
Comment (georgiosd) [inline engi-server/Types/Analysis/AnalysisMutations.cs:48]: The right way to do this is to add `DisableGithubIntegration` in `EngiOptions` and set the default value to `false` in `appsettings.json`, then use `bool.Parse` to convert to boolean before checking if the flag is there.

**Review by georgiosd — CHANGES_REQUESTED** [review]

**Comment by garrettmaring** [comment]
Patching the DB for integration tests elsewhere sounds better than the source changes here. @cck197 and I discussed making a GH apps & users for tests as well.

**Comment by georgiosd** [comment]
Okay, are we closing this? Want me to make a sample bash file or whatever with the patch?

**Comment by garrettmaring** [comment]
Let's wait for @cck197 reply

**Comment by cck197** [comment]
Patching the user in RavenDB works! Thank you.

Closing.

**Pull request closed without merging**